libs/components/src/lib/indoor-maps.component.ts

```diff
@@ -0,0 +1,494 @@
+                }
+            } else {
+                this.geolocation_error_message =
+                    'Error: geolocation is not supported.';
+                reject('Geolocation not supported');
+            }
+        });
+    }
+
+    private _updateGeolocation(updated_location: GeolocationPosition) {
+        if (updated_location) {
+            if (
+                updated_location.coords?.latitude !== this.user_latitude ||
+                updated_location.coords?.longitude !== this.user_longitude
+            ) {
+                this.user_latitude = updated_location.coords?.latitude;
+                this.user_longitude = updated_location.coords?.longitude;
+                this.getRoute(this.selected_destination);
+            }
+        }
+    }
+
+    private _handleGeolocationError(error: any) {
+        this._alertError('Error updating your geolocation.');
+    }
+
+    getRoute(location: any) {
+        this.selected_destination = location;
+        if (this.user_latitude && this.user_longitude) {
+            const originLocationCoordinate: any = {
+                lat: this.user_latitude,
+                lng: this.user_longitude,
+            };
+
+            //Hardcoded coordinates for mock map in Austin
+            // const originLocationCoordinate = {
+            //     lat: 30.3603774,
+            //     lng: -97.7426772,
+            // };
+
+            const destinationCoordinate = {
+                lat: location.properties.anchor.coordinates[1],
+                lng: location.properties.anchor.coordinates[0],
+                floor: location.properties.floor,
+            };
+
+            const routeParameters = {
+                origin: originLocationCoordinate,
+                destination: destinationCoordinate,
+                travelMode: 'WALKING',
+            };
+
+            this.mapsIndoors_directions_service_instance
+                .getRoute(routeParameters)
+                .then((directionsResult: any) => {
+                    this.mapsIndoors_directions_renderer_instance?.setRoute(
+                        directionsResult
+                    );
+                })
+                .catch((error: any) => {
+                    console.error('Error fetching route: ' + error);
+                    if (
+                        error instanceof TypeError &&
+                        error.message?.includes('origin')
+                    ) {
+                        this.route_error_message =
+                            'Error: Cannot create route as origin location is outside of map area.';
+                        this._alertError(this.route_error_message);
+                    }
+                });
+        } else {
+            this.route_error_message = 'Error: unable to find a route.';
+            this._alertError(this.route_error_message);
+        }
+    }
+
+    private _alertError(error_message: string) {
+        this._snackBar.open(error_message, 'OK', {
```

Comment:
Use the `notifyError` method from the `@placeos/common` lib.

Reply:
I've now updated this.